cmd2/cmd2.py

```diff
@@ -424,8 +433,13 @@ def __init__(self, completekey: str = 'tab', stdin=None, stdout=None, persistent
         # Used load command to store the current script dir as a LIFO queue to support _relative_load command
         self._script_dir = []
 
-        # Used when piping command output to a shell command
-        self.pipe_proc = None
+        # A flag used to protect critical sections in the main thread from stopping due to a KeyboardInterrupt
```

Comment:
Interesting/weird.  It is a context manager which is also treated as a boolean flag within the sigint handler.  I would recommend rewording the comments associated with this.  Maybe something like:
> A context manager used to protect critical sections in the main thread from stopping due to a KeyboardInterrupt